### PR TITLE
Feature/109 password recovery email input

### LINKED ISSUE #109: Password Recovery - Email input
- Implement password recovery for local email/password users
- This involves a screen that lets local users enter their e-mail address to retrieve a password recovery e-mail with a link in order to reset their password.
- User should also be required to answer a captcha challenge (to cut down on security issues and spamming)

To do:
- A page with an input box for the user's email
- Implement a captcha challenge that is required to be completed before the request is submitted.
- Submit button
- Implement functionality that sends a request to the API to start the password recovery process
- API responds with OK or Error. OK will result in user being prompted to check their e-mail for recovery instructions.

### PR DESCRIPTION
**Pull request by davideastmond**
This resolves #109 . 
PR contains:
- UI implementation for the password recovery request feature on the front end.
- Implements Google recaptcha v2 API
- Functionality to send the user's request and recaptcha response data to the API for processing.
- Handles the API response / error.

### CODE REVIEW

File: app/src/index.css [hunk 1]
```diff
@@ -27,7 +27,7 @@
   --color-white: #ffffff;
   --color-black: #000000;
 
-  --error-color: var(--color-red);
+  --error-color: var(--color-pale);
```
Comment (lily-law) [inline app/src/index.css]: Maybe we could use a more vibrant tone, this one is a little mellow so doesn't standout enough IMO

File: app/src/pages/password-recovery/password-recovery.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+const loadjs = require("loadjs");
+function PasswordRecovery() {
+  const [email, setEmail] = useState<string>("");
+  const [emailErrorMessage, setEmailErrorMessage] = useState<string>("");
+  const [requestSuccessfullySubmitted, setRequestSuccessfullySubmitted] = useState<boolean>(false);
+  loadjs("https://www.google.com/recaptcha/api.js");
+
+  const sendPasswordRecoveryRequest = async(e:any) => {
+    
+    assert(process.env.REACT_APP_DEV_CAPTCHA_SITE_KEY, "Please set env variable REACT_APP_DEV_CAPTCHA_SITE_KEY")
```
Comment (lily-law) [inline app/src/pages/password-recovery/password-recovery.tsx]: :100: good idea putting this in

**Review by lily-law — APPROVED** [review]
:+1: Looks good, will review the backend PR's for this soon

File: app/src/index.css (hunk 1 above)
Comment (davideastmond) [reply app/src/index.css]: Yeah I was playing around with this. I found the red difficult to see against the green background. I'll play around with this.

Comment (lily-law) [reply app/src/index.css]: Ah yes I hadn't thought of that, in the design it was only tested against the black background. Maybe the orange #D9532C would work nice?

File: app/src/index.css [hunk 3]
```diff
@@ -27,7 +27,7 @@
   --color-white: #ffffff;
   --color-black: #000000;
 
-  --error-color: var(--color-pale);
+  --error-color: var(--color-red);
```
Comment (lily-law) [inline app/src/index.css]:
```suggestion
  --error-color: var(--color-orange-0);
```

**Review by lily-law — APPROVED** [review]
:+1:

File: app/src/index.css (hunk 3 above)
Comment (davideastmond) [reply app/src/index.css]: I updated it to orange but it's still a bit difficult to read. How about if we add a black background color to the h4 element?

Comment (lily-law) [reply app/src/index.css]: Yes that can work, better that it's legible and we can always review later to improve. I think the only two pages with the green background are the signup page and the password recovery so we could just add the styling to those.

**Review by lily-law — APPROVED** [review]
:+1:

**Pull request merged**